### DISCUSSION TITLE
Parcel as React Library bundler

### DISCUSSION THREAD

**Discussion by diego-hourly** [parcel-bundler/parcel#7495] [Q&A] [answered]
I've bee trying to use the setup suggested by the website for libraries (https://parceljs.org/getting-started/library/) but including react in the process to create a React library component. But keep getting errors. First 
```
> parcel watch

🚨 Build failed.

@parcel/core: Failed to resolve '@parcel/transformer-react-refresh-wrap/lib/helpers/helpers.js' from
'./node_modules/@parcel/transformer-react-refresh-wrap/lib/ReactRefreshWrapTransformer.js'

@parcel/resolver-default: External dependency "@parcel/transformer-react-refresh-wrap" is not declared in package.json.

```

And after including `@parcel/transformer-react-refresh-wrap` as a dependency 
```
> parcel watch

🚨 Build failed.

@parcel/packager-js: Asset was skipped or not found.

  AssertionError [ERR_ASSERTION]: Asset was skipped or not found.
  at ScopeHoistingPackager.getSymbolResolution
```

Not sure why is not able to parse jsx in watch mode.

Here are the 2 files I have in place

```json
{
  "name": "@poc/commons",
  "version": "1.0.0",
  "description": "",
  "source": "src/index.js",
  "module": "dist/module.js",
  "scripts": {
    "watch": "parcel watch",
    "build": "parcel build"
  },
  "devDependencies": {
    "parcel": "^2.0.1"
  },
  "dependencies": {
    "@parcel/transformer-react-refresh-wrap": "^2.0.1"
  },
  "peerDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}

```

```javascript
// src/index.js

export const Button = ({ children }) => <button>{children}</button>;
```

NOTE: running `parcel build` works. It's watch mode the one that fails

**Comment by diego-hourly** [comment, marked as answer]
It was my fault. Parcel doesn't support "Arrow function expressions" by default. Including a `.babelrc` with 

``` json
{
  "presets": ["react-app"]
}
```

solves my issue.

**Comment by e-krebs** [comment]
FYI, I filed a bug on that issue, as we shouldn't have to use `babel` to build a React library: https://github.com/parcel-bundler/parcel/issues/7652

**Reply by mattmogford** [reply]
Did the big get resolved? I'm getting this exact issue today
